examples/vae/utils.py

```diff
@@ -0,0 +1,90 @@
+"""
+    This code is created with reference to torchvision/utils.py.
+
+    Modify: torch.tensor -> jax.numpy.DeviceArray
+
+    If you want to know about this file in detail, please visit the original code:
+        https://github.com/pytorch/vision/blob/master/torchvision/utils.py
+"""
+import math
+import jax
+import jax.numpy as np
+
+irange = range
+
+
+def make_grid(ndarray, nrow=8, padding=2,
+              normalize=False, range=None, scale_each=False, pad_value=0.0):
+    """Make a grid of images.
+    Args:
+        ndarray (array_like or list): 4D mini-batch ndarray of shape (B x H x W x C)
+            or a list of images all of the same size.
+        nrow (int, optional): Number of images displayed in each row of the grid.
+            The final grid size is ``(B / nrow, nrow)``. Default: ``8``.
+        padding (int, optional): amount of padding. Default: ``2``.
+        normalize (bool, optional): If True, shift the image to the range (0, 1),
+            by the min and max values specified by :attr:`range`. Default: ``False``.
+        range (tuple, optional): tuple (min, max) where min and max are numbers,
+            then these numbers are used to normalize the image. By default, min and max
+            are computed from the tensor.
+        scale_each (bool, optional): If ``True``, scale each image in the batch of
+            images separately rather than the (min, max) over all images. Default: ``False``.
+        pad_value (float, optional): Value for the padded pixels. Default: ``0``.
+    """
+    if not (isinstance(ndarray, np.DeviceArray) or
+            (isinstance(ndarray, list) and all(isinstance(t, np.DeviceArray) for t in ndarray))):
+        raise TypeError('array_like or list of tensors expected, got {}'.format(type(ndarray)))
+
+    # if list of tensors, convert to a 4D mini-batch ndarray
+    if isinstance(ndarray, list):
+        ndarray = np.stack(ndarray, 0)
+
+    if ndarray.ndim == 2:  # single image H x W
```

Comment:
this function accepts many more input signatures than the docstring provides. I think it is better to just support B,H,W,C case

Reply:
I just removed the supports except for B,H,W,C case.